pytorch_lightning/metrics/functional/psnr.py

```diff
@@ -31,21 +60,28 @@ def psnr(
     data_range: Optional[float] = None,
     base: float = 10.0,
     reduction: str = 'elementwise_mean',
+    dim: Optional[Union[int, Tuple[int, ...]]] = None,
 ) -> torch.Tensor:
     """
     Computes the peak signal-to-noise ratio
 
     Args:
         preds: estimated signal
         target: groun truth signal
-        data_range: the range of the data. If None, it is determined from the data (max - min)
+        data_range:
+            the range of the data. If `None`, it is determined from the data (max - min). `data_range` must be given
+            when `dim` is not `None`. (default: `None`)
         base: a base of a logarithm to use (default: 10)
         reduction: a method to reduce metric score over labels.
```

Comment:
```suggestion
        reduction: a string indicating how to reduce individual metric scores.
```

Reply:
Will the change make the doc different from other metrics? For example,

- https://github.com/PyTorchLightning/pytorch-lightning/blob/ba806c8ee0ff9a038471e151d703e38f357d234a/pytorch_lightning/metrics/functional/ssim.py#L133
- https://github.com/PyTorchLightning/pytorch-lightning/blob/ba806c8ee0ff9a038471e151d703e38f357d234a/pytorch_lightning/metrics/functional/iou.py#L89

Reply:
You are right, it is probably better to keep it consistent between different metrics. Keep it as it is